plugins/tiledb/io/TileDBWriter.cpp

```diff
@@ -0,0 +1,345 @@
+* AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+* OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT
+* OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY
+* OF SUCH DAMAGE.
+****************************************************************************/
+
+#include <limits>
+
+#include <boost/any.hpp>
+#include <pdal/util/FileUtils.hpp>
+
+#include "TileDBWriter.hpp"
+
+namespace pdal {
+
+static PluginInfo const s_info
+{
+    "writers.tiledb",
+    "Write data using TileDB.",
+    "http://pdal.io/stages/drivers.tiledb.writer.html"
+};
+
+CREATE_SHARED_STAGE(TileDBWriter, s_info)
+
+template <typename T>
+void setAttrAnyValue(std::map<std::string, pdalboost::any>& mp, const std::string& key, const T val)
+{
+    auto it = mp.find(key);
+    if (it == mp.end())
+    {
+        auto sp = std::make_shared<std::vector<T>>();
+        mp[key] = sp;
+    }
+    auto sp = pdalboost::any_cast<std::shared_ptr<std::vector<T>>>(mp[key]);
+    sp->push_back(val);
+}
+
+void writeAttributeValue(const std::string& attrName,
+        const PointViewPtr view, const PointId idx, std::map<std::string, pdalboost::any>& mp)
+{
+    Dimension::Id id = Dimension::id(attrName);
+    Dimension::Type t = view->dimType(id);
+
+    switch (t)
+    {
+        case Dimension::Type::Double:
+            setAttrAnyValue(mp, attrName, view->getFieldAs<double>(id, idx));
+            break;
+        case Dimension::Type::Float:
+            setAttrAnyValue(mp, attrName, view->getFieldAs<float>(id, idx));
+            break;
+        case Dimension::Type::Signed8:
+            setAttrAnyValue(mp, attrName, view->getFieldAs<char>(id, idx));
+            break;
+        case Dimension::Type::Signed16:
+            setAttrAnyValue(mp, attrName, view->getFieldAs<short>(id, idx));
+            break;
+        case Dimension::Type::Signed32:
+            setAttrAnyValue(mp, attrName, view->getFieldAs<int>(id, idx));
+            break;
+        case Dimension::Type::Signed64:
+            setAttrAnyValue(mp, attrName, view->getFieldAs<long>(id, idx));
+            break;
+        case Dimension::Type::Unsigned8:
+            setAttrAnyValue(mp, attrName, view->getFieldAs<unsigned char>(id, idx));
+            break;
+        case Dimension::Type::Unsigned16:
+            setAttrAnyValue(mp, attrName, view->getFieldAs<unsigned short>(id, idx));
+            break;
+        case Dimension::Type::Unsigned32:
+            setAttrAnyValue(mp, attrName, view->getFieldAs<unsigned int>(id, idx));
+            break;
+        case Dimension::Type::Unsigned64:
+            setAttrAnyValue(mp, attrName, view->getFieldAs<unsigned long>(id, idx));
+            break;
+        case Dimension::Type::None:
+        default:
+            throw pdal_error("Unsupported attribute type for " + view->dimName(id));
+    }
+}
+
+tiledb::Attribute createAttribute(const tiledb::Context& ctx, const std::string name, Dimension::Type t)
```

Comment:
Should we use a default compressor for the attributes?

Reply:
:+1: Added compression options to the writer